wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java

```diff
@@ -194,6 +201,29 @@
 	 */
 	public ApiConnection(String apiBaseUrl) {
 		this.apiBaseUrl = apiBaseUrl;
+		this.cookies = new HashMap<>();
+	}
+	
+	/**
+	 * Deserializes an existing ApiConnection from JSON.
+	 * 
+	 * @param apiBaseUrl
+	 * 		base URL of the API to use, e.g. "https://www.wikidata.org/w/api.php/"
+	 * @param cookies
+	 * 		map of cookies used for this session
+	 * @param loggedIn
+	 * 		true if login succeeded.
+	 */
+	@JsonCreator
+	protected ApiConnection(
```

Comment:
What about making it package-private? It seems cleaner than protected.

Reply:
Why not! When refactoring the datamodel I have often found myself reusing these constructors at other places in the package so I generally mark these as protected, but I don't have this need in this particular case.